python/src/aiconfig/scripts/aiconfig_cli.py

```diff
@@ -116,6 +139,24 @@ def _run_editor_servers(edit_config: EditServerConfig) -> Result[list[str], str]
 
 def _process_cli_config(cli_config: AIConfigCLIConfig) -> Result[bool, str]:
     LOGGER.setLevel(cli_config.log_level)
+
+    config_path = cli_config.config_path
+    file_contents = yaml.dump({"allow_usage_data_sharing": cli_config.allow_usage_data_sharing})
+    try:
+        with open(config_path, "x") as f:
+            f.write(file_contents)
+    except FileExistsError:
+        with open(config_path, "r") as f:
+            existing_file_contents = f.read()
```

Comment:
I get what you're doing, but I feel "x" is pretty niche, and it's more intuitive to do 

1. Read first from file
2. If FileNotFoundError, create a new one with the "w" command

Reply:
Why does this cause a race condition?

Reply:
This is #869, but the race condition is:
1. check that file exists, it does not
2. File gets written
3. Overwrite file unintentionally

Reply:
> Overwrite file unintentionally

How come this overwrites the file untintentionally if it doesn't exist before?

Reply:
This is unrelated to 870. I'll move this to 869.

Reply:
Questions answered in https://github.com/lastmile-ai/aiconfig/pull/869#discussion_r1450809345, makes sense even though it's an edge case